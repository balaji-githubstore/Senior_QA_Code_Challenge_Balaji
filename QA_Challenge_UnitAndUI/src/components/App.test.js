import App from "./App";
import '@testing-library/jest-dom'
import { render, screen } from "@testing-library/react";
import React from 'react';
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'
import { VisibilityFilters } from '../actions'
import ReactDOMServer from 'react-dom/server';

const middlewares = []
const mockStore = configureStore(middlewares)

//mocking store with todos and visibility filter
const store = mockStore({
    todos: [{ id: 1, text: 'Buy groceries1', completed: false }],
    visibilityFilter: VisibilityFilters.SHOW_ALL
})

describe("App Unit Test", () => {

    //verify div tag
    it("should render div tag", () => {
        const app = App()
        expect(app.type).toBe('div');
    })

    //verify AddTodo connect function
    it("should render AddTodo under div tag", () => {
        const app = App()
        expect(app.props.children[0]).toMatchSnapshot("<AddTodo />")
    })

    //verify TodoList connect function
    it("should render TodoList under div tag", () => {
        const app = App()
        expect(app.props.children[1]).toMatchSnapshot("<TodoList />")
    })

      //verify Footer connect function
    it("should render Footer under div tag", () => {
        const app = App()
        expect(app.props.children[2]).toMatchSnapshot("<Footer />")
    })

    //verify all button
    it("should render all button", () => {
        render(
            <Provider store={store}>
                <App />
            </Provider>
        )
        const buttons = screen.getAllByRole(/button/i)
        expect(buttons[0]).toHaveTextContent('Add Todo')
        expect(buttons[1]).toHaveTextContent('All')
        expect(buttons[2]).toHaveTextContent('Active')
        expect(buttons[3]).toHaveTextContent('Completed')
    })

     //verify one textbox
    it("should render one textbox", () => {
        render(
            <Provider store={store}>
                <App />
            </Provider>
        )
        const textbox = screen.getByRole(/textbox/i)
        expect(textbox).toBeInTheDocument()
    })

})