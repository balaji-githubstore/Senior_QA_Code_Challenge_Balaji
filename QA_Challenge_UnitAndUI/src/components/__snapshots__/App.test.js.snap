// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Unit Test should : <body>
        
              <li
                style="text-decoration: none;"
              >
                Buy groceries1
              </li>
            </
          </div>
        </div>
      </body> 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div>
          <form>
            <input />
            <button
              type="submit"
            >
              Add Todo
            </button>
          </form>
        </div>
        <ul>
          <li
            style="text-decoration: none;"
          >
            Buy groceries1
          </li>
        </ul>
        <div>
          <span>
            Show: 
          </span>
          <button
            disabled=""
            style="margin-left: 4px;"
          >
            All
          </button>
          <button
            style="margin-left: 4px;"
          >
            Active
          </button>
          <button
            style="margin-left: 4px;"
          >
            Completed
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div>
        <form>
          <input />
          <button
            type="submit"
          >
            Add Todo
          </button>
        </form>
      </div>
      <ul>
        <li
          style="text-decoration: none;"
        >
          Buy groceries1
        </li>
      </ul>
      <div>
        <span>
          Show: 
        </span>
        <button
          disabled=""
          style="margin-left: 4px;"
        >
          All
        </button>
        <button
          style="margin-left: 4px;"
        >
          Active
        </button>
        <button
          style="margin-left: 4px;"
        >
          Completed
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`App Unit Test should : <body>
        <div>
          <div>
            <div>
              <form>
                <input />
                <button
                  type="submit"
                >
                  Add Todo
                </button>
              </form>
            </div>
            <ul>
              <li
                style="text-decoration: none;"
              >
                Buy groceries1
              </li>
            </ul>
            <div>
              <span>
                Show:
              </span>
              <button
                disabled=""
                style="margin-left: 4px;"
              >
                All
              </button5>
              <button
                style="margin-left: 4px;"
              >
                Active
              </button>
              <button
                style="margin-left: 4px;"
              >
                Completed
              </button>
            </div>
          </div>
        </div>
      </body> 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div>
          <form>
            <input />
            <button
              type="submit"
            >
              Add Todo
            </button>
          </form>
        </div>
        <ul>
          <li
            style="text-decoration: none;"
          >
            Buy groceries1
          </li>
        </ul>
        <div>
          <span>
            Show: 
          </span>
          <button
            disabled=""
            style="margin-left: 4px;"
          >
            All
          </button>
          <button
            style="margin-left: 4px;"
          >
            Active
          </button>
          <button
            style="margin-left: 4px;"
          >
            Completed
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div>
        <form>
          <input />
          <button
            type="submit"
          >
            Add Todo
          </button>
        </form>
      </div>
      <ul>
        <li
          style="text-decoration: none;"
        >
          Buy groceries1
        </li>
      </ul>
      <div>
        <span>
          Show: 
        </span>
        <button
          disabled=""
          style="margin-left: 4px;"
        >
          All
        </button>
        <button
          style="margin-left: 4px;"
        >
          Active
        </button>
        <button
          style="margin-left: 4px;"
        >
          Completed
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`App Unit Test should render AddTodo under div tag: <AddTodo /> 1`] = `<UNDEFINED />`;

exports[`App Unit Test should render Footer under div tag: <Footer /> 1`] = `<Footer />`;

exports[`App Unit Test should render TodoList under div tag: <TodoList /> 1`] = `<UNDEFINED />`;
